model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2119,3 +2136,162 @@ async def execute(self, user: User, request: ModelDownloadRequest) -> ModelDownl
             public_file_name = model_file.rsplit("/", 1)[-1]
             urls[public_file_name] = self.filesystem_gateway.generate_signed_url(model_file)
         return ModelDownloadResponse(urls=urls)
+
+
+class CreateBatchCompletionsUseCase:
+    def __init__(
+        self,
+        docker_image_batch_job_gateway: DockerImageBatchJobGateway,
+        docker_repository: DockerRepository,
+        docker_image_batch_job_bundle_repo: DockerImageBatchJobBundleRepository,
+    ):
+        self.docker_image_batch_job_gateway = docker_image_batch_job_gateway
+        self.docker_repository = docker_repository
+        self.docker_image_batch_job_bundle_repo = docker_image_batch_job_bundle_repo
+
+    def infer_hardware_from_model_name(
+        self, model_name: str
+    ) -> CreateDockerImageBatchJobResourceRequests:
```

Comment:
Gah I just realized this should be `CreateDockerImageBatchJobResourceRequest`, oh well.

Reply:
possible to rename, but don't want to put in this PR